cluster/src/main/java/org/apache/iotdb/cluster/utils/RaftUtils.java

```diff
@@ -302,4 +317,106 @@ public static Status createErrorStatus(String errorMsg){
+    hashNodeMap.entrySet().forEach(entry -> res.put(entry.getKey(), entry.getValue().getIp()));
+    return res;
+  }
+
+  public static Map<Integer, String> getVirtualRing() {
+    SortedMap<Integer, VirtualNode> hashNodeMap = router.getVirtualRing();
+    Map<Integer, String> res = new LinkedHashMap<>();
+    hashNodeMap.entrySet().forEach(entry -> res.put(entry.getKey(), entry.getValue().getPhysicalNode().getIp()));
+    return res;
+  }
+
+  /**
+   * Get all node information of the data group of input storage group.
+   * The first node is the current leader
+   *
+   * @param sg storage group ID. If null, return metadata group info
+   */
+  public static PeerId[] getDataPartitionOfSG(String sg) {
+    String groupId;
+    PeerId[] nodes;
+    if (sg == null) {
+      groupId = ClusterConfig.METADATA_GROUP_ID;
+      nodes = (PeerId[]) ((RaftService) server.getMetadataHolder().getService()).getPeerIdList()
+          .toArray();
+    } else {
+      PhysicalNode[] group = router.routeGroup(sg);
+      groupId = router.getGroupID(group);
+      nodes = getPeerIdArrayFrom(group);
+    }
+    PeerId leader = RaftUtils.getLeaderPeerID(groupId);
+    for (int i = 0; i < nodes.length; i++) {
+      if (leader.equals(nodes[i])) {
+        PeerId t = nodes[i];
+        nodes[i] = nodes[0];
+        nodes[0] = t;
+        break;
+      }
+    }
+    return nodes;
+  }
+
+  public static Map<String[], String[]> getDataPartitionOfNode(String ip) {
+    return getDataPartitionOfNode(ip, config.getPort());
+  }
+
+  public static Map<String[], String[]> getDataPartitionOfNode(String ip, int port) {
+    PhysicalNode[][] groups = router.getGroupsNodes(ip, port);
+
+    Map<String, List<String>> groupSGMap = new LinkedHashMap<>();
+    for (int i = 0; i < groups.length; i++) {
+      groupSGMap.put(generateStringKey(groups[i]), new ArrayList<>());
+    }
+    Set<String> allSGList = ((MetadataStateManchine)((RaftService)server.getMetadataHolder().getService()).getFsm()).getAllStorageGroups();
+    for (String sg : allSGList) {
+      String key = generateStringKey(router.routeGroup(sg));
+      if (groupSGMap.containsKey(key)) {
+        groupSGMap.get(key).add(sg);
+      }
+    }
+
+    String[][] groupIps = new String[groups.length][];
+    for (int i = 0; i < groups.length; i++) {
+      groupIps[i] = new String[groups[i].length];
+      for (int j = 0; j < groups[i].length; j++) {
+        groupIps[i][j] = groups[i][j].ip;
+      }
+    }
+
+    Map<String[], String[]> res = new HashMap<>();
+    int index = 0;
+    for (Entry<String, List<String>> entry : groupSGMap.entrySet()) {
+      res.put(groupIps[index], entry.getValue().toArray(new String[entry.getValue().size()]));
+      index++;
+    }
+    return res;
+  }
+
+  private static String generateStringKey(PhysicalNode[] nodes) {
+    if (nodes == null || nodes.length == 0) {
+      return "";
+    }
+    Arrays.sort(nodes, new Comparator<PhysicalNode>() {
```

Comment:
Please simplify this ` Arrays.sort` method to make code more concise

Reply:
Change to `Comparator.comparing(PhysicalNode::toString)`